qemu/tests/virtio_scsi_mq.py

```diff
@@ -143,24 +133,17 @@ def proc_interrupts_results(results):
         raise error.TestError("Didn't find addr from qtree. Please check "
                               "the log.")
     error.context("Check device init status in guest", logging.info)
-    init_check_cmd = params.get("init_check_cmd", "dmesg | grep irq")
-    output = session.cmd_output(init_check_cmd)
-    irqs_pattern = params.get("irqs_pattern", "%s:\s+irq\s+(\d+)")
-    irqs_pattern = irqs_pattern % scsi_bus_addr
-    irqs_watch = re.findall(irqs_pattern, output)
-    # As there are several interrupts count for virtio device:
-    # config, control, event and request. And the each queue have
-    # a request count. So the totally count for virtio device should
-    # equal to queus number plus three.
-    if len(irqs_watch) != 3 + int(num_queues):
-        raise error.TestFail("Failed to check the interrupt ids from dmesg")
     irq_check_cmd = params.get("irq_check_cmd", "cat /proc/interrupts")
     output = session.cmd_output(irq_check_cmd)
-    irq_results, _ = proc_interrupts_results(output)
-    for irq_watch in irqs_watch:
-        if irq_watch not in irq_results:
-            raise error.TestFail("Can't find irq %s from procfs" % irq_watch)
-
+    logging.info('The info of interrupters before testing :\n%s ' % output)
+    irq_name = params.get("irq_regex")
+    if irq_name:
+        prev_irq_results, _, irqs_watch = proc_interrupts_results(output, irq_name)
+    cpu_select = 1
+    for irq_id in irqs_watch:
+        bind_cpu_cmd = "echo %s > /proc/irq/%s/smp_affinity" % (re.split("0x", hex(cpu_select))[1:][0], irq_id)
```

Comment:
Looks `hex(cpu_select).replace('0x', '')` is better.

Reply:
ok, i will update it